pybamm/models/lithium_ion/dfn.py

```diff
@@ -0,0 +1,126 @@
+#
+# Doyle-Fuller-Newman (DFN) Model
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pybamm
+import numpy as np
+
+
+class DFN(pybamm.BaseModel):
+    """Doyle-Fuller-Newman (DFN) model of a lithium-ion battery.
+
+    Attributes
+    ----------
+
+    rhs: dict
+        A dictionary that maps expressions (variables) to expressions that represent
+        the rhs
+    initial_conditions: dict
+        A dictionary that maps expressions (variables) to expressions that represent
+        the initial conditions
+    boundary_conditions: dict
+        A dictionary that maps expressions (variables) to expressions that represent
+        the boundary conditions
+    variables: dict
+        A dictionary that maps strings to expressions that represent
+        the useful variables
```

Comment:
These are no longer the only attributes of the model. For ease of maintenance, we should avoid repeating attributes in the model docstrings and just say which model it extends